tests/test_datacite.py

```diff
@@ -170,10 +169,10 @@ def test_datacite_transform_with_all_fields_transforms_correctly(
 
 
 def test_datacite_transform_missing_required_datacite_fields_logs_warning(caplog):
-    input_records = parse_xml_records(
+    source_records = Datacite.parse_source_file(
```

Comment:
Just a note: as we discussed, if the getting of corrrect transformer class and parsing of input records is combined, I think a lot of these explicit calls to specific transformers could be replaced with something like:

```python
transformer = Transformer.load("datacite", "file_path.xml")
```

But, also acknowledging this is likely for a future PR.

Reply:
It's a subtle thing, but if we decide to move more functionality into the `Transformer` class, we might want to not refer to an instantiated `Transformer` as `output_records`.

Then again, if we keep as a true iterator in essence, I can the logic behind that variable naming.